common4j/src/main/com/microsoft/identity/common/java/cache/MsalOAuth2TokenCache.java

```diff
@@ -367,10 +366,11 @@ public ICacheRecord save(@NonNull final GenericOAuth2Strategy oAuth2Strategy,
 
         // Save the Account and Credentials...
         saveAccounts(accountToSave);
-        saveCredentialsInternal(accessTokenToSave, refreshTokenToSave, idTokenToSave);
-
-        // Remove old refresh tokens (except for the one we just saved) if it's MRRT or FRT
-        removeAllRefreshTokensExcept(accountToSave, refreshTokenToSave);
+        synchronized(this) {
```

Comment:
You probably want to synchronize this on a static monitor rather than an instance.